### PR TITLE
Seabeta 566 change your answers warning

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: SeaPublicWebsite.BusinessLogic/Models/PropertyData.cs [hunk 1]
```diff
@@ -45,6 +45,8 @@ public class PropertyData
     public int? HoursOfHeatingEvening { get; set; }
     public decimal? Temperature { get; set; }
     public PropertyData UneditedData { get; set; }
+    public bool? HasSeenRecommendations { get; set; }
+    public bool? HasEditedData { get; set; }
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite.BusinessLogic/Models/PropertyData.cs:49]: HasSeenRecommendations is set to true when the user clicks 'See Results' on Answer Summary. Maybe I can rename this to 'HasConfirmedAsnwers' or 'RecommendationsRequested'?

HasEditedData is set to true from the beginning, and when the user comes in the answer summary page with property data different than the Unedited Data (i.e. a change occured).
It is set to false after requesting the recommendations.

Comment (DanCorderSoftwire) [reply SeaPublicWebsite.BusinessLogic/Models/PropertyData.cs:49]: These don't need to be nullable?
`HasSeenRecommendations` seems straightforward.
For `HasEditedData` could we start it as `true`, set it to `false` when we request recommendations from BRE and only set it to `true` when we call in `DeleteUneditedData` in `UpdatePropertyAndRedirect` (if something has actually changed).
That seems simpler and I think covers all the cases?

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

File: SeaPublicWebsite.UnitTests/PropertyDataTests.cs [hunk 2]
```diff
@@ -64,13 +66,54 @@ public void CopiesAllAnswers()
...
+        // Assert
+        propertyData.PropertyRecommendations.Should().BeNullOrEmpty();
+    }
+    
+    public void CommitEditsKeepsPropertyRecommendationsIfThereAreNoChange()
+    {
+        // Arrange
+        var propertyData = InitializePropertyData();
+        propertyData.CreateUneditedData();
+        propertyData.WallConstruction = null;
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite.UnitTests/PropertyDataTests.cs:111]: This is a change, so the test should fail?

File: SeaPublicWebsite.BusinessLogic/Models/PropertyData.cs [hunk 3]
```diff
@@ -56,26 +55,31 @@ public void CreateUneditedData()
...
+    private bool EditedDataIsDifferent()
     {
-        if (UneditedData is null)
-        {
-            HasEditedData = true;
-            return;
-        }
-
-        HasEditedData = OwnershipStatus != UneditedData.OwnershipStatus
+        return OwnershipStatus != UneditedData.OwnershipStatus
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite.BusinessLogic/Models/PropertyData.cs:82]: How will we notice if someone forgets to update this method?

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

**Pull request merged**